WordPress/php-toolkit
# Run PHPUnit on PHP >= 7.0

### What does this PR do?
- refactors GitHub's actions to run a matrix of tests for PHP >= 7.0
- implements [PHPUnit-Polyfills](https://github.com/Yoast/PHPUnit-Polyfills)
- reorganizes test files for ease of access and extension
- simplifies composers, exposing one slick file for library users
- separates concerns, grouping dev-related files in the `dev` directory

### What problem does it fix?
- tests are now run for all required PHP versions
- there were 2 top-level composer files, now only the right one is exposed

### How to test if it works?
- create a branch and set up a PR, the CI should run and test your code against all PHP & OS versions
- expect at least 27 checks
